lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1823,162 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    id_searches, type_searches, checkin_searches, searches = [], [], [], []
+
+    # Group search terms by what's being searched for
+    terms.each do |term|
+      case term.split(":").first
+      when "session_id"
+        id_searches << term
+      when "session_type"
+        type_searches << term
+      when "last_checkin"
+        checkin_searches << term
+      else
+        print_error("Please provide valid search term. Given: #{term.split(":").first}")
+        return
+      end
+    end
+
+    # Make sure user is properly querying checkin
+    unless checkin_searches.length() < 2
+      unless validate_checkin_searches(checkin_searches)
+        return
+      end
+    end
```

Comment:
you should validate the checkin searches regardless of how many you have since if you only have a single checkin search you'll miss out on the rest of the validation you have (like that only `before` and `after` are valid)

Reply:
that validation occurs later, but you're right it'd be good to consolidate it into that function to clean up the logic!

Reply:
Circling back - `checkin_searches.length() < 2` is itself a validation, so the logic does need to be tweaked but we need to keep that line _somewhere_

Reply:
isn't that validation redundant since if you have 2 or more searches you'' either have an invalid operator _or_ a duplicate? but if you think it's necessary then can't it be moved inside the `validate_checkin_searches` function?